jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -595,7 +595,7 @@ protected boolean passConditionalHeaders(HttpServletRequest request, HttpServlet
                     return false;
                 }
 
-                long ifmsl = request.getDateHeader(HttpHeader.IF_MODIFIED_SINCE.asString());
+                long ifmsl = DateParser.parseDate(ifms);
```

Comment:
Per https://www.rfc-editor.org/rfc/rfc9110#section-13.1.3-6

> A recipient MUST ignore the If-Modified-Since header field if the received field value is not a valid HTTP-date, the field value has more than one member, or if the request method is neither GET nor HEAD.

If `ifmsl` is `-1` indicating a bad Date, we should ignore the rest of the handling of `If-Modified-Since`.